constraints/src/main/java/gov/nasa/jpl/aerie/constraints/tree/AssignGaps.java

```diff
@@ -0,0 +1,56 @@
+package gov.nasa.jpl.aerie.constraints.tree;
+
+import gov.nasa.jpl.aerie.constraints.model.EvaluationEnvironment;
+import gov.nasa.jpl.aerie.constraints.model.Profile;
+import gov.nasa.jpl.aerie.constraints.model.SimulationResults;
+import gov.nasa.jpl.aerie.constraints.time.Interval;
+
+import java.util.Objects;
+import java.util.Set;
+
+public final class AssignGaps<P extends Profile<P>> implements Expression<P> {
+  public final Expression<P> originalProfile;
+  public final Expression<P> defaultProfile;
+
+  public AssignGaps(final Expression<P> originalProfile, final Expression<P> defaultProfile) {
+    this.originalProfile = originalProfile;
+    this.defaultProfile = defaultProfile;
+  }
+
+  @Override
+  public P evaluate(final SimulationResults results, final Interval bounds, final EvaluationEnvironment environment) {
+    final var originalProfile = this.originalProfile.evaluate(results, bounds, environment);
+    final var defaultProfile = this.defaultProfile.evaluate(results, bounds, environment);
+
+    return originalProfile.assignGaps(defaultProfile);
+  }
+
+  @Override
+  public void extractResources(final Set<String> names) {
+    this.originalProfile.extractResources(names);
+    this.defaultProfile.extractResources(names);
+  }
+
+  @Override
+  public String prettyPrint(final String prefix) {
+    return String.format(
+        "\n%s(assignGaps %s %s)",
+        prefix,
+        this.originalProfile.prettyPrint(prefix + "  "),
+        this.defaultProfile.prettyPrint(prefix + "  ")
+    );
+  }
+
+  @Override
+  public boolean equals(final Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    AssignGaps<?> that = (AssignGaps<?>) o;
+    return Objects.equals(originalProfile, that.originalProfile) && Objects.equals(defaultProfile, that.defaultProfile);
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
If this were a
```java
record AssignGaps<P extends Profile<P>>(Expression<P> originalProfile, Expression<P> defaultProfile) implements Expression<P> {
  ...
}
```
then `hashCode()` and `equals()` would be autogenerated for you.